docs/src/utils/track.ts

```diff
@@ -159,3 +163,60 @@ export const trackSearchQuery = (
   }
   window.location.assign(suggestion.url);
 };
+
+interface Event {
+  name: string;
+  type: string;
+}
+
+export const track = (event: Event, data?: Record<string, any>) => {
+  try {
+    const opt = {
+      event,
+      data,
+    };
+    globalThis.AWSMA.ready(() => {
+      document.dispatchEvent(
+        new CustomEvent(globalThis.AWSMA.TRIGGER_EVENT, { detail: opt })
+      );
+    });
+  } catch (error) {
+    // don't want to error for analytics events
+  }
+};
+
+export const trackCopy = (code: string) => {
+  track(
+    {
+      type: 'click',
+      name: 'CopyCode',
+    },
+    {
+      code,
+    }
+  );
+};
+
+export const trackHomeCodeEdit = () => {
+  track({
```

Comment:
shouldn't this use the trackClick function?

Reply:
Yup and moved it out of this file since it is homepage specific